liblepton/src/scheme_object.c

```diff
@@ -719,6 +721,80 @@ SCM_DEFINE (set_object_color_x, "%set-object-color!", 2, 0, 0,
   return obj_s;
 }
 
+
+
+/*! \brief Check whether an object is locked.
+ *
+ * \par Function Description
+ * Check the state of an object's selectable flag: if it's true, the
+ * object is considered to be unlocked, otherwise it is locked.
+ *
+ * \note Scheme API: Implements the %object-locked? procedure in the
+ * (geda core object) module.
+ *
+ * \param obj_s  #OBJECT smob to inspect.
+ *
+ * \return       Boolean value indicating whether \a obj_s is locked.
+ */
+SCM_DEFINE (object_locked_x, "%object-locked?", 1, 0, 0,
+            (SCM obj_s), "Check whether an object is locked.")
+{
+  SCM_ASSERT (EDASCM_OBJECTP (obj_s), obj_s,
+              SCM_ARG1, s_object_locked_x);
+
+  OBJECT* obj = edascm_to_object (obj_s);
+
+  return obj->selectable ? SCM_BOOL_F : SCM_BOOL_T;
+
+} /* object_locked_x() */
+
+
+
+/*! \brief Lock or unlock an object.
+ *
+ * \par Function Description
+ * Set object's selectable flag: locked objects cannot be selected.
+ *
+ * \note Scheme API: Implements the %set-object-locked! procedure in
+ * the (geda core object) module.
+ *
+ * \param obj_s     #OBJECT smob to modify.
+ * \param locked_s  boolean: whether the object should be locked.
+ *
+ * \return          the object (\a obj_s).
+ */
+SCM_DEFINE (set_object_locked_x, "%set-object-locked!", 2, 0, 0,
+            (SCM obj_s, SCM locked_s), "Lock or unlock an object.")
+{
+  SCM_ASSERT (EDASCM_OBJECTP (obj_s), obj_s,
+              SCM_ARG1, s_set_object_locked_x);
+  SCM_ASSERT (scm_is_bool (locked_s), locked_s,
+              SCM_ARG2, s_set_object_locked_x);
+
+  OBJECT* obj = edascm_to_object (obj_s);
+
+  int locked = locked_s == SCM_BOOL_T;
```

Comment:
Just one note. In Scheme boolean world, any non-`#f` value of a variable is `true`. Your expression `int locked = locked_s == SCM_BOOL_T;` makes this... ehm... a bit contrariwise, everything non-`#t` gets `false`, which is wrong, I believe. You would better use the `scm_is_true()` Scheme function in this case.

Reply:
8-\ Didn't know about that. Absolutely counter-intuitive. And just against the common sense, I think. :)

Reply:
As in C, any non-zero value is true. What's wrong? :-)